featureDevelopmentGotchas.md

```diff
@@ -0,0 +1,25 @@
+# Edge Case Spidey Sense Tips and Tricks - Things to keep in mind for code development/reviews:
+(aka this is the kind of shit that has bitten us in the past)
+
+## Future proof?
+- Are you building in convention instead of design?
+  - How easy would it be for an unfamiliar dev to use your code improperly?
+
+## Complete logic
+- Are there unaccounted for logical branches?
+  - Wherever there is an if, consider the else
+  - Use exhaustive switch cases, always.
+  - DO NOT PUT IN EMPTY CASES OR SWALLOW ERRORS - log at a minimum
+  - Kotlin specific - If we do the work inside of a ?.let{} then what happens if the object IS null? Did you think about that?!
```

Comment:
What do you think about having some convention when we talk about languages that we can use throughout the site? Not necessarily a code block, but maybe something formatted like `Kotlin`: xxxx